catalyst/dl/callbacks/gan.py

```diff
@@ -53,10 +53,10 @@ def _get(dictionary: dict, keys: Optional[Union[str, List[str]]]) -> Any:
 
     def on_batch_end(self, state: State):
         """On batch end call"""
-        outputs = self._get(state.output, self.output_key)
-        targets = self._get(state.input, self.input_key)
+        outputs = self._get(state.batch_out, self.output_key)
```

Comment:
We can now remove this entire callback (see TODO above - it is now done) and make `WassersteinDistanceCallback` descendant of `MetricCallback`

Reply:
For now I prefer to leave GAN updated up to you :)
Core refactoring first and all other improvement after that... or it will just never end